### PR TITLE
Introduce verify_kdumpcrash_smp_force_no_rebuild test case in P4 section

### PR DESCRIPTION
**Pull request by 0xba1a**
Introducing a new kdump test to verify the default behavior of CBL-Mariner-2.0 - `verify_kdumpcrash_smp_force_no_rebuild`.

The default behavior of CBL-Mariner-2.0 is set `force_no_rebuild` to `1`. So, a new initrd will not be generated for kdump kernel and the kdump kernel will use the main initrd itself. New distributions including AzureLinux-3.0 started generating a separate initrd for the kdump kernel as a default behavior. So, all the kdump-crash tests are regenerating new initrd. As we still need to support CBL-Mariner-2.0, we are in a need to verify the old behavior too. So, introducing this new test which is currently applicable only to CBL-Mariner-2.0.

### CODE REVIEW

File: lisa/tools/kdump.py [hunk 1]
```diff
@@ -194,6 +194,10 @@ class KdumpBase(Tool):
 
     dump_path = "/var/crash"
 
+    kdump_config = {
```
Comment (0xba1a) [inline lisa/tools/kdump.py:197]: KdumpBase is a tool, and it is shared across the test-cases running in parallel. So, if it gets modified in one test-case, its affecting all the other test-cases. So, the current implementation is wrong.

@squirrelsc  @LiliDeng Can you please point me towards some reference where I can maintain node / test-case specific config parameters?

Comment (LiliDeng) [reply lisa/tools/kdump.py:197]: @0xba1a pass parameter in https://github.com/microsoft/lisa/blob/a6361265df2c47fd44785c1288d956ad61e3d31f/microsoft/testsuites/kdump/kdumpcrash.py#L341

**Comment by 0xba1a** [comment]
AMD64 Test results

```
2024-11-06 15:40:16.358[133877338048256][INFO] lisa.RootRunner  KdumpCrash.verify_kdumpcrash_smp_force_no_rebuild: PASSED   
2024-11-06 15:40:16.358[133877338048256][INFO] lisa.RootRunner         KdumpCrash.verify_kdumpcrash_on_random_cpu: PASSED   
2024-11-06 15:40:16.359[133877338048256][INFO] lisa.RootRunner                   KdumpCrash.verify_kdumpcrash_smp: PASSED   
2024-11-06 15:40:16.359[133877338048256][INFO] lisa.RootRunner           KdumpCrash.verify_kdumpcrash_single_core: SKIPPED  deployment skipped: ["no available quota found on 'westus3'."], runbook: Environment(name='generated_3', topology='subnet', nodes_raw=None, nodes_requirement=[type:requirement,name:,default:False,count:1,core:1,mem:[2048,],disk:has_resource_disk: None,os_disk_type: allowed:True,items:[DiskType.Ephemeral,DiskType.StandardHDDLRS,DiskType.StandardSSDLRS,DiskType.PremiumSSDLRS],os_disk_size: [0,],data_disk_type: allowed:True,items:[DiskType.StandardHDDLRS,DiskType.PremiumV2SSDLRS,DiskType.UltraSSDLRS,DiskType.StandardSSDLRS,DiskType.PremiumSSDLRS],count: [0,],caching: None,iops: [0,],throughput: [0,],size: [0,],max_data_disk_count: [0,],disk_controller_type: allowed:False,items:[DiskControllerType.NVME,DiskControllerType.SCSI],network interface: data_path:allowed:False,items:[NetworkDataPath.Synthetic,NetworkDataPath.Sriov], nic_count:[1,], max_nic_count:[1,], gpu:[0,],f:allowed:True,items:[SecurityProfileSettings(extended_schemas={}, type='Security_Profile', security_profile=allowed:False,items:[SecurityProfileType.Standard,SecurityProfileType.SecureBoot], encrypt_disk=False, disk_encryption_set_id=''),gen:2,arch:x64],ef:None,NodeSpace(type='requirement')], _original_nodes_requirement=[type:requirement,name:,default:False,count:1,core:1,mem:[2048,],disk:has_resource_disk: None,os_disk_type: allowed:True,items:[DiskType.Ephemeral,DiskType.StandardHDDLRS,DiskType.StandardSSDLRS,DiskType.PremiumSSDLRS],os_disk_size: [0,],data_disk_type: allowed:True,items:[DiskType.StandardHDDLRS,DiskType.PremiumV2SSDLRS,DiskType.UltraSSDLRS,DiskType.StandardSSDLRS,DiskType.PremiumSSDLRS],count: [0,],caching: None,iops: [0,],throughput: [0,],size: [0,],max_data_disk_count: [0,],disk_controller_type: allowed:False,items:[DiskControllerType.NVME,DiskControllerType.SCSI],network interface: data_path:allowed:False,items:[NetworkDataPath.Synthetic,NetworkDataPath.Sriov], nic_count:[1,], max_nic_count:[1,], gpu:[0,],f:allowed:True,items:[SecurityProfileSettings(extended_schemas={}, type='Security_Profile', security_profile=allowed:False,item
2024-11-06 15:40:16.360[133877338048256][INFO] lisa.RootRunner              KdumpCrash.verify_kdumpcrash_on_cpu32: PASSED   
2024-11-06 15:40:16.360[133877338048256][INFO] lisa.RootRunner test result summary
2024-11-06 15:40:16.361[133877338048256][INFO] lisa.RootRunner     TOTAL    : 5
2024-11-06 15:40:16.361[133877338048256][INFO] lisa.RootRunner     QUEUED   : 0
2024-11-06 15:40:16.362[133877338048256][INFO] lisa.RootRunner     ASSIGNED : 0
2024-11-06 15:40:16.362[133877338048256][INFO] lisa.RootRunner     RUNNING  : 0
2024-11-06 15:40:16.362[133877338048256][INFO] lisa.RootRunner     FAILED   : 0
2024-11-06 15:40:16.363[133877338048256][INFO] lisa.RootRunner     PASSED   : 4
2024-11-06 15:40:16.363[133877338048256][INFO] lisa.RootRunner     SKIPPED  : 1
```

**Comment by 0xba1a** [comment]
Closing this as the same issue fixed differently in another PR - https://github.com/microsoft/lisa/pull/3498

**Pull request closed without merging**